### ISSUE TITLE
you should be able to run `go run . local` without flags to emulate a `kubefirst local` without flags

### ISSUE THREAD

**Issue by johndietz** [kubefirst/kubefirst#1113]
tl;dr: main should mean main everywhere if nothing directed the cli otherwise.

task: we should have a tight correlation between the `kubefirst` and `go .` syntaxes.

fully recognize that this is a nuanced issue, we should plan to meet to discuss the implications on this one if it's not all 👍 

the context:
### binaries
- the kubefirst cli binaries that we produce have default tags of our upstream github repositories like kubefirst/gitops-template. to describe the purpose, kubefirst cli version 1.1.0 should be bound to the gitops-template (and other -template) repos of that same 1.1.0 tag by default. they were built with each other and that makes sense. we are doing this part well.
- we offer to our users that they can fork our upstream gitops-template repository and reference their copy during installation, this is possible today by including any combination of gitops-owner, repo, and branch flags. these flags could use cleanup and better description/visualization, but we're functionally able to do this now - no known changes needed on this part.
### source
- developers need to be in branches of both the cli repo (kubefirst/kubefirst) and template repos like gitops-template in order to make full platform adjustments and test the stack thoroughly. since in this case we are running from source there is no ldflag set for the version/tag, and there are flags available `--gitops-branch` and `--metaphor-branch` to override this value anyway. whether an ldflag for version/tag is set or not, if `--gitops-branch` or `--metaphor-branch` is provided, it should be honored as the preference over tags/version. i think we're doing this, but this need reconfirmation at conclusion of any changes to this logic.
- new contributors need to be in the main branch of kubefirst/kubefirst and run the experience against main everywhere as a default. we have a bug where running `go run . local` with no flags fails with a panic on the push only to metaphor-frontend, the gitops repo is fine. but metaphor-frontend panics on fallback logic. 

due to that last source bullet, the proposal is to have logic so that the push to metaphor-frontend defaults to main if (a) an ldflag is not set for version and (b) the branch flags mentioned above were not used. 

the order of precedence proposed:
always wins: specified flag owner, specified flag branch
next winner: ldflags baked in at release used to pull from kubefirst owner and tags for template repos
if neither ldflags are set, nor owner is set: kubefirst is the owner, main branch is the branch pulled from all template repos

**Comment by johndietz** [comment]
this is how to replicate the problem area we need to solve:
current behavior - we tell users in the kubefirst readme that we can emulate `kubefirst` with `go run .` but when you run `go run . local` today:
- it fails because you didn’t provide `--metaphor-branch` and panics on push to new github origin
- when you then run `go run . local destroy` to recover, the command actually succeeds, BUT it fails to remove the 2 git repos it created
- then when you `go run . clean` and run with the flag you collide with the repos that never deleted and have to destroy again to recover, then clean, then if committed enough, what are the flags, etc

**Comment by converge** [comment]
> fully recognize that this is a nuanced issue, we should plan to meet to discuss the implications on this one if it's not all 👍

if we want to move the discussion for the next release, this PR is doing the job to fix the flag requirement: https://github.com/kubefirst/kubefirst/pull/1120

@johndietz wdyt?

**Comment by johndietz** [comment]
@converge yeah this looks like the functionality im after. any issues including it for 1.11.0?

**Comment by converge** [comment]
> @converge yeah this looks like the functionality im after. any issues including it for 1.11.0?

awesome! 

I spent some time testing different possibilities[1], we can include it for 1.11.0.

[1] with dev. mode code, using binary, binary with ldflag set, adding parameters, without parameters, everything is working as expected for these different cases,
